### PR TITLE
try fix open hand bug #463

### PR DESCRIPTION
**Pull request by unStatiK**
possible fix for standart tenhou rules (for part of rules  with immediality reveal dora after ankan)

### CODE REVIEW

File: Mimir/src/helpers/onlineLog/Parser.php [hunk 1]
```diff
@@ -67,11 +69,35 @@ public function parseToSession(SessionPrimitive $session, string $content, $with
         $reader = new \XMLReader();
         $reader->XML($content);
 
+        $kanNakiCache = null;
+        $this->_ankan_cache = array(
+            0 => array(),
+            1 => array(),
+            2 => array(),
+            3 => array());
```
Comment (ctizen) [inline Mimir/src/helpers/onlineLog/Parser.php:77]: 1) please use camel case
2) `array()` can be replaced by `[]`, please use `[]` for consistency with other codebase.
3) array indices can be omitted when starting from 0
```
$this->_ankanCache = [[], [], [], []];
```

File: Mimir/src/helpers/onlineLog/Parser.php [hunk 2]
```diff
@@ -50,6 +50,8 @@ class OnlineParser
      */
     protected $_lastTokenIsAgari = false;
 
+    protected $_ankan_cache = [];
```
Comment (ctizen) [inline Mimir/src/helpers/onlineLog/Parser.php:53]: Please make sure to run `make check` locally to run all code style tools, linters and tests. Absence of phpdoc comment for class member should output static analysis error.

File: Mimir/src/helpers/onlineLog/Parser.php [hunk 3]
```diff
@@ -67,11 +69,35 @@ public function parseToSession(SessionPrimitive $session, string $content, $with
...
+
         while ($reader->read()) {
             if ($reader->nodeType != \XMLReader::ELEMENT) {
                 continue;
             }
 
+            if ($reader->localName === 'N') {
+                $meld = $reader->getAttribute('m');
+                $nakiWho = $reader->getAttribute('who');
+                $kanNakiCache = array('m' => $meld, 'who' => $nakiWho);
```
Comment (ctizen) [inline Mimir/src/helpers/onlineLog/Parser.php:87]: same with `[]` here

File: Mimir/src/helpers/onlineLog/Parser.php [hunk 4]
```diff
@@ -67,11 +69,35 @@ public function parseToSession(SessionPrimitive $session, string $content, $with
...
+            }
+            else if ($reader->localName !== 'DORA') {
```
Comment (ctizen) [inline Mimir/src/helpers/onlineLog/Parser.php:89]: This looks like code style error, but this one can be fixed with `make autofix`

File: Mimir/src/helpers/onlineLog/Parser.php [hunk 5]
```diff
@@ -365,6 +404,11 @@ protected function _tokenAGARI(\XMLReader $reader): void
     // round start, reset all needed things
     protected function _tokenINIT(): void
     {
+        $this->_ankan_cache = array(
+            0 => array(),
+            1 => array(),
+            2 => array(),
+            3 => array());
```
Comment (ctizen) [inline Mimir/src/helpers/onlineLog/Parser.php:411]: Same comment as in constructor

**Review by ctizen — CHANGES_REQUESTED** [review]

File: Mimir/src/helpers/onlineLog/Parser.php (hunk 1 above)
Comment (unStatiK) [reply Mimir/src/helpers/onlineLog/Parser.php]: fixed

File: Mimir/src/helpers/onlineLog/Parser.php (hunk 2 above)
Comment (unStatiK) [reply Mimir/src/helpers/onlineLog/Parser.php]: fixed

File: Mimir/src/helpers/onlineLog/Parser.php (hunk 4 above)
Comment (unStatiK) [reply Mimir/src/helpers/onlineLog/Parser.php]: fixed

File: Mimir/src/helpers/onlineLog/Parser.php (hunk 5 above)
Comment (unStatiK) [reply Mimir/src/helpers/onlineLog/Parser.php]: fixed

File: Mimir/src/helpers/onlineLog/Parser.php (hunk 3 above)
Comment (unStatiK) [reply Mimir/src/helpers/onlineLog/Parser.php]: fixed

**Review by ctizen — APPROVED** [review]

**Pull request merged**